beacon-chain/sync/validate_attester_slashing.go

```diff
@@ -0,0 +1,58 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/karlseguin/ccache"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+)
+
+// seenAttesterSlashings represents a cache of all the seen slashings
+var seenAttesterSlashings = ccache.New(ccache.Configure())
+
+func attSlashingCacheKey(slashing *ethpb.AttesterSlashing) (string, error) {
+	hash, err := hashutil.HashProto(slashing)
+	if err != nil {
+		return "", err
+	}
+	return string(hash[:]), nil
+}
+
+// Clients who receive an attester slashing on this topic MUST validate the conditions within VerifyAttesterSlashing before
+// forwarding it across the network.
+func (r *RegularSync) validateAttesterSlashing(ctx context.Context, msg proto.Message, p p2p.Broadcaster) bool {
+	slashing, ok := msg.(*ethpb.AttesterSlashing)
+	if !ok {
+		return false
+	}
+	cacheKey, err := attSlashingCacheKey(slashing)
+	if err != nil {
+		log.WithError(err).Warn("could not hash attester slashing")
+		return false
+	}
+
+	if seenAttesterSlashings.Get(cacheKey) != nil {
+		return false
+	}
+
+	state, err := r.db.HeadState(ctx)
+	if err != nil {
+		log.WithError(err).Error("Failed to get head state")
+		return false
+	}
+
+	if err := blocks.VerifyAttesterSlashing(state, slashing); err != nil {
+		log.WithError(err).Warn("Received invalid attester slashing")
```

Comment:
Where do we blacklist failed attester slashing? 

Say we got to this point, the attester slashing failed to process, we error'ed out, and the same attester slashing comes again